tests/api_test.py

```diff
@@ -45,85 +46,137 @@
 config.parse_flags_with_absl()
 FLAGS = config.FLAGS
 
-class APITest(jtu.JaxTestCase):
-
-  def test_grad_argnums(self):
-    def f(x, y, z, flag=False):
-      assert flag
-      return 1.0 * x + 2.0 * y + 3.0 * z
-
-    assert grad(f)(1.0, 1.0, 1.0, flag=True) == 1.0
-    assert grad(f, argnums=1)(1.0, 1.0, 1.0, flag=True) == 2.0
-    assert grad(f, argnums=(2, 0))(1.0, 1.0, 1.0, flag=True) == (3.0, 1.0)
-
-  def test_value_and_grad_argnums(self):
-    def f(x, y, z, flag=False):
-      assert flag
-      return 1.0 * x + 2.0 * y + 3.0 * z
 
-    y = f(1.0, 1.0, 1.0, flag=True)
-    assert api.value_and_grad(f)(1.0, 1.0, 1.0, flag=True) == (y, 1.0)
-    assert api.value_and_grad(f, argnums=1)(1.0, 1.0, 1.0, flag=True) == (y, 2.0)
-    assert api.value_and_grad(f, argnums=(2, 0))(1.0, 1.0, 1.0, flag=True) == (y, (3.0, 1.0))
-
-  def test_jit_static_args(self):
+class JitTest(jtu.JaxTestCase):
```

Comment:
There are many tests that exercise `api.jit`, what should be the criteria by which they should be added to `JitTest` vs `ApiTest`?

Reply:
In practice,  2 comments here:
- first, is that this refactoring will be "necessary" when introducing the C++ _cpp_jit (See https://github.com/google/jax/pull/4051/files#diff-df85584b1d7860fe4ea0efa9236f180e) because we want to run the same tests on both the Python tests + C++ tests (and while the Python tests will have more features, there will be a few more functions). Using a class with the jit tests make it easier to create another one, extending the first one.
- second, for specifically your question, for me, tests that involve the test of the code of jax.jit should be there. In particular, I will also later move the `donate_argnums` tests to JitTest. I am not expecting tests that do not test jax.jit, but *uses* jax.jit for a test to be there.

So we should have here tests for the different arguments, scalar and numpy array support, donage_argnums, static_argnums, disable_jit etc. 

To test that a while-loop can be jitted, I would not put it into the JitTest but into the test that tests the while-loop.

Reply:
Sounds good. I was thikning that there are really a lot of tests that test compilation, e.g., all the tests that compare jitted code with numpy code. I think that it will be somewhat obvious what tests can go in JitTest, once you add the comparison tests with C++ Jit. 

Perhaps it would help to add a couple of line comments on how people should use JitTest, e.g., what tests go in there